modin/pandas/base.py

```diff
@@ -2864,6 +2864,23 @@ def tz_localize(
         )
         return self.set_axis(labels=new_labels, axis=axis, inplace=not copy)
 
+    def value_counts(
+        self,
+        subset: Sequence[Hashable] = None,
+        normalize: bool = False,
+        sort: bool = True,
+        ascending: bool = False,
+        dropna: bool = True,
+    ):
+        if subset is None:
+            subset = self._query_compiler.columns
+        counted_values = self.groupby(by=subset, sort=not sort, dropna=dropna).size()
+        if sort:
+            counted_values.sort_values(ascending=ascending, inplace=True)
```

Comment:
It would be good to sort values during reduce phase. Can we do it?

Reply:
Unfortunately, we can't. The whole idea of moving `value_counts` to the front-end is to implement it in a natural way via GroupBy. However, groupby API doesn't have an option to sort _values_ of the resulted object, that's why we can't fit handling of the `sort=True` (as well as `normalize=True`) in a single query as it was done before.

This problem is described at the [PR's header](https://github.com/modin-project/modin/pull/2730#issue-571857342). The possible solution is to bring some kind of query optimizer for a pandas backend, which would pack all of these function calls into a single backend call.

Reply:
What current performance of `value_counts` is against pandas? This front-end based implementation would be good to experience that query optimizer but not sure we should keep such things in the front-end if they are significantly slower than pandas. At least until the query optimizer is implemented.

Reply:
@YarShev posted expanded timing comparison [here](https://github.com/modin-project/modin/pull/2730#issuecomment-902703098)